packages/core/useOnceInAllTabs/index.md

```diff
@@ -0,0 +1,21 @@
+---
+category: Browser
+---
+
+# useOnceInAllTabs
+
+Uses the [Web Locks API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Locks_API) to elect one tab to do work.
+
+Use this composable to e.g. refresh a session only once even when multiple tabs of your application are open.
+
+## Usage
+
+```ts
+import { useOnceInAllTabs } from '@vueuse/core'
+
+const { doOnce, isElected, isSupported } = useOnceInAllTabs({ name: 'lock-name' })
```

Comment:
maybe we should go for: `useWebLocks()` instead of `useOnceInAllTabs()` 🤔

Reply:
You are totally right, the name `useOnceInAllTabs` is not the best name. The Web Locks API has multiple use cases. Leader election is only one of them. IMHO, when we name the composable `useWebLocks` it should also handle the other use cases and features of the Web Locks API (shared locks, lock stealing, non-blocking lock requests). I did not figure out a good way to do that, tho.

What about the name `useWebLocksLeader` (and renaming `doOnce` to `asLeader` or `asLeaderDo`)?

Reply:
I think I need to take a closer look at Web Locks. I have never used them before.

As described in: https://github.com/vueuse/vueuse/discussions/4548#discussioncomment-12115352

IMO we should add `useWebLocks`. `useOnceInAllTabs` (or whatever the name is) should wrap `useWebLocks`. So if we add other use cases we will have a common ground at `useWebLocks` and increase visibility with related composables.

Reply:
@OrbisK I think I have found a reasonably flexible way for a `useWebLocks` composable. Please take a look at it.